backend/src/main/java/com/zpi/backend/reservations/Reservation.java

```diff
@@ -0,0 +1,85 @@
+package com.zpi.backend.reservations;
+
+import com.zpi.backend.game_instance.GameInstance;
+import com.zpi.backend.reservation_status.ReservationStatus;
+import com.zpi.backend.reservations.DTO.NewReservationDTO;
+import com.zpi.backend.user.User;
+import com.zpi.backend.utils.DateUtils;
+import jakarta.persistence.*;
+import lombok.Data;
+
+import java.util.Date;
+import java.util.UUID;
+
+@Entity
+@Data
+@Table(name = "reservations")
+public class Reservation {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+
+    @Column(name = "reservation_id",nullable = true,unique = true)
+    private String reservationId;
+
+
+    @ManyToOne
+    private User renter;
+
+
+    @Column(name="start_date",nullable = false)
+    private Date startDate;
+
+    @Column(name="end_date",nullable = false)
+    private Date endDate;
+
+    @ManyToOne
+    private ReservationStatus status;
+
+    @ManyToOne
+    private GameInstance gameInstance;
+
+    @Column(name="renter_comment",length = 500)
+    private String renterComment;
+
+    @Column(name="timestamp")
+    private Date timestamp;
+
+    @Transient
+    private int duration;
+
+
+
+
+    public Reservation() {
+
+    }
+
+    public Reservation fromDTO(NewReservationDTO newReservationDTO, User renter, GameInstance gameInstance) {
+
+        this.renter = renter;
+        this.gameInstance = gameInstance;
+        this.startDate = newReservationDTO.getStartDate();
+        this.endDate = newReservationDTO.getEndDate();
+        this.renterComment = newReservationDTO.getRenterComment();
+        this.timestamp = new Date(System.currentTimeMillis());
+        this.duration = DateUtils.getDuration(startDate,endDate);
+        this.reservationId = UUID.randomUUID().toString();
```

Comment:
Wasn't supposed to be ReservationNo and year-month-id?

Reply:
it was and it is set like that in addReservation in reservationService but there is a need for an initial value and i thought that uuid is better then something like null

Reply:
Ok. So let's leave it like it is now.